src/index.ts

```diff
@@ -322,17 +326,31 @@ function main(): void {
 
   cacheAction.idslib.onMain(async () => {
     if (cacheAction.noopMode) {
-      actionsCore.warning(NOOP_TEXT);
+      actionsCore.warning(TEXT_NOOP);
       return;
     }
 
+    if (cacheAction.idslib.nixStoreTrust === "untrusted") {
+      actionsCore.warning(TEXT_TRUST_UNTRUSTED);
```

Comment:
Do we really want to exit with a warning instead of just erroring? If the runner user isn't trusted, then there's no point in having this action, and if that were me, I'd want to be notified that this is what happened (i.e. I personally would expect an error).

Reply:
Philosophically I look at it like where we only warn and go into noop mode if Nix isn't installed. I'd rather not force the user to delete the action altogether if the trust situation is just temporary.